edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/PurchaseResult.cs

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Http
+{
+    using System;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Hub.Core.Billing;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+
+    public abstract class PurchaseResult
+    {
+        protected PurchaseResult(HttpStatusCode statusCode)
+        {
+            this.StatusCode = statusCode;
+        }
+
+        [JsonIgnore]
+        public HttpStatusCode StatusCode { get; }
+    }
+
+    public class PurchaseResultSuccess : PurchaseResult
+    {
+        public PurchaseResultSuccess()
+            : base(HttpStatusCode.OK)
+        {
+        }
+
+        [JsonConverter(typeof(StringEnumConverter))]
+        [JsonProperty("purchaseStatus")]
+        public PurchaseStatus PurchaseStatus { get; set; }
+
+        [JsonProperty("publisherId")]
```

Comment:
This seems redundant with the "SynchedPurchase" class. Why not return this class instead of those 3 info?